tachyon/math/polynomials/univariate/icicle/icicle_ntt_bn254.cc

```diff
@@ -18,3 +24,95 @@ cudaError_t tachyon_bn254_ntt_cuda(const ::bn254::scalar_t* input, int size,
 cudaError_t tachyon_bn254_release_domain(::device_context::DeviceContext& ctx) {
   return ::ntt::release_domain<::bn254::scalar_t>(ctx);
 }
+
+namespace tachyon::math {
+
+template <>
+bool IcicleNTT<bn254::Fr>::Init(const bn254::Fr& group_gen,
+                                const IcicleNTTOptions& options) {
+#if FIELD_ID != BN254
+#error Only BN254 is supported
+#endif
+  ::device_context::DeviceContext ctx{stream_, /*device_id=*/0, mem_pool_};
+  math::BigInt<4> group_gen_big_int = group_gen.ToBigInt();
+  // TODO(chokobole): There are some issues about domain initialization, we need
+  // to handle each of these.
```

Comment:
Please change in all related files
```suggestion
  // TODO(chokobole): We must handle these issues with domain initialization: